src/harmony.c

```diff
@@ -0,0 +1,118 @@
+#include <stdbool.h>
+#include <stdint.h>
+#include <os.h>
+#include <cx.h>
+#include "harmony.h"
+#include "bech32.h"
+
+
+void deriveOneKeypair(cx_ecfp_private_key_t *privateKey, cx_ecfp_public_key_t *publicKey) {
+	uint8_t keySeed[32];
+	cx_ecfp_private_key_t pk;
+
+	// bip32 path for ONE is 44'/1023'/0'/0/0
+	uint32_t bip32Path[] = {44 | 0x80000000, 1023 | 0x80000000,  0x80000000, 0, 0};
+    os_perso_derive_node_bip32(CX_CURVE_256K1, bip32Path, 5, keySeed, NULL);
+    cx_ecfp_init_private_key(CX_CURVE_256K1, keySeed, 32, &pk);
```

Comment:
Why are we doing this work in case when both privateKey, publicKey are both NULL